File: Spa/ClientApp/src/app/models/CandidateRace.ts
import { Candidate } from './candidate';

export class CandidateRace {
  candidateRaceId: number;
  positionName: string;
  platformInfo: string;
  topIssues: string;
  candidateId: number;
  raceId: number;
  candidate: Candidate
  constructor(
    candidateRaceId: number,
    positionName: string,
    platformInfo: string,
    topIssues: string,
    candidateId: number,
    raceId: number,
    candidate: Candidate
  ) {
    this.candidateRaceId = candidateRaceId;
    this.candidateId = candidateId;
    this.positionName = positionName;
    this.platformInfo = platformInfo;
    this.topIssues = topIssues;
    this.raceId = raceId;
    this.candidate = candidate;
  }
}

File: VotingModelLibrary/Models/CandidateRace.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VotingModelLibrary.Models
{
    public class CandidateRace
    {
        [Key]
        [Display(Name = "CandidateRaceId")]
        public int CandidateRaceId { get; set; }
        [Display(Name = "PositionName")]
        public string PositionName { get; set; }
        [Display(Name = "PlatformInfo")]
        public string PlatformInfo { get; set; }
        [Display(Name = "TopIssues")]
        public string TopIssues { get; set; }
        [Display(Name = "CandidateId")]
        public int CandidateId { get; set; }
        [Display(Name = "Candidate")]
        public Candidate Candidate { get; set; }

        [Display(Name = "RaceId")]
        public int? RaceId { get; set; }
        [Display(Name = "Race")]
        public virtual Race Race { get; set; }
    }
}
